### ISSUE TITLE
terracotta connect requesting metadata keys in wrong order

### ISSUE THREAD

**Issue by rumski20** [DHI-GRAS/terracotta#314]
When I run `terracotta connect` on an instance of terracotta server (eg. `localhost:5000`), I get a bunch of 404 errors in the console:
```
XHRGET
http://localhost:5000/metadata/cc/d30/WHNGS/2020
[HTTP/1 404 NOT FOUND 10ms]
```
http://localhost:5000/metadata/d30/cc/2020/WHNGS is the correct order as indicated by http://localhost:5000/keys

I can probably fix this locally, if you can point me to where these `/metadata` requests are being generated.

**Comment by dionhaefner** [comment]
Ah, this is what I suspected to be the cause of #305 and #310. Good to see it confirmed!

@DenizYil do you think you could take a stab at this? I remember you mentioning that you felt confident to do work on the client.

**Comment by DenizYil** [comment]
Yes, absolutely! I noticed it too and was planning on fixing it along with the other changes to the client 😄

**Comment by dionhaefner** [comment]
@DenizYil I was planning to release terracotta 8.1 as soon as possible. Do you think we could add this fix, potentially without the other changes to the client?

**Comment by dionhaefner** [comment]
Unfortunately not.

I'm contemplating whether we should rip out the react app and go back to the bare-bones vanilla Javascript client I wrote by hand. That one I can at least maintain myself. I see little value in keeping components that provide virtually no functional improvements and are not actively maintained.

**Comment by DenizYil** [comment]
I'd like to really apologize for the delay with the PR. For context, I started my master's on the 1st and have severely cut the number of hours I'm working. I also completely forgot about the PR until I received a notification for this, which is completely on me. I'll look at the errors now and hopefully, we should be able to merge it today or tomorrow. 

Sincere apologies for the delay.

> Unfortunately not.
> 
> I'm contemplating whether we should rip out the react app and go back to the bare-bones vanilla Javascript client I wrote by hand. That one I can at least maintain myself. I see little value in keeping components that provide virtually no functional improvements and are not actively maintained.
This could definitely be an avenue we could consider. I haven't seen the old bare-bones client, but I'm a bit against it considering that after merging the PR, the client should be good for a long time. I would also like to put extra effort into maintaining it moving forward if that is of course okay with you.

Something I was thinking about was if we could potentially host this on a website instead of having it in this repository. DHI could easily host this on Azure, and that would allow us to move the code to another open-source repository. I was going to propose it, but then I realized it wouldn't really be beneficial for localhost Terracotta instances, which I'm not sure how we could handle.
